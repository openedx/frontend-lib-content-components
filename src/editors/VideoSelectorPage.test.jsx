import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';
import VideoSelectorPage from './VideoSelectorPage';

const props = {
  blockId: 'block-v1:edX+DemoX+Demo_Course+type@html+block@030e35c4756a4ddc8d40b95fbbfff4d4',
  courseId: 'course-v1:edX+DemoX+Demo_Course',
  lmsEndpointUrl: 'evenfakerurl.com',
  studioEndpointUrl: 'fakeurl.com',
};

jest.mock('react-redux', () => ({
  Provider: 'Provider',
  connect: (mapStateToProps, mapDispatchToProps) => (component) => ({
    mapStateToProps,
    mapDispatchToProps,
    component,
  }),
}));
jest.mock('./VideoSelector', () => 'VideoSelector');

describe('Video Selector Page', () => {
  describe('snapshots', () => {
    test('rendering correctly with expected Input', () => {
      expect(shallow(<VideoSelectorPage {...props} />).snapshot).toMatchSnapshot();
    });
    test('rendering with props to null', () => {
      expect(shallow(<VideoSelectorPage />).snapshot).toMatchSnapshot();
    });
  });
});
