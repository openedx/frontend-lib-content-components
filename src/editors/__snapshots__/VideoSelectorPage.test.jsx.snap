// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Selector Page snapshots rendering correctly with expected Input 1`] = `
<Provider
  store={
    {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <ErrorBoundary
    learningContextId="course-v1:edX+DemoX+Demo_Course"
    studioEndpointUrl="fakeurl.com"
  >
    <VideoSelector
      blockId="block-v1:edX+DemoX+Demo_Course+type@html+block@030e35c4756a4ddc8d40b95fbbfff4d4"
      learningContextId="course-v1:edX+DemoX+Demo_Course"
      lmsEndpointUrl="evenfakerurl.com"
      studioEndpointUrl="fakeurl.com"
    />
  </ErrorBoundary>
</Provider>
`;

exports[`Video Selector Page snapshots rendering with props to null 1`] = `
<Provider
  store={
    {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <ErrorBoundary
    learningContextId={null}
    studioEndpointUrl={null}
  >
    <VideoSelector
      blockId={null}
      learningContextId={null}
      lmsEndpointUrl={null}
      studioEndpointUrl={null}
    />
  </ErrorBoundary>
</Provider>
`;
